velox/exec/tests/UnnestTest.cpp

```diff
@@ -101,6 +133,141 @@ TEST_F(UnnestTest, arrayWithOrdinality) {
+  vector_size_t size = array1.size();
+  auto arrayVectorDict = makeArrayOfRowVector(rowType1, array1);
+  auto reversedIndices = makeIndicesInReverse(size);
+  vector = makeRowVector(
+      {wrapInDictionary(reversedIndices, size, arrayVectorDict),
+       BaseVector::wrapInConstant(size, 0, arrayVector2)});
+
+  op = PlanBuilder().values({vector}).unnest({}, {"c0", "c1"}).planNode();
+  expected = makeRowVector(
+      {makeNullableFlatVector<int32_t>({1, 2, std::nullopt, 3, 4, 5, 6}),
+       makeNullableFlatVector<StringView>(
+           {"red", "blue", std::nullopt, "green", "orange", "black", "white"}),
+       makeNullableFlatVector<int64_t>({-1, -2, -3, -1, -2, -3, std::nullopt}),
+       makeNullableFlatVector<double>(
+           {1.0, 2.0, 3.0, 1.0, 2.0, 3.0, std::nullopt}),
+       makeNullableFlatVector<StringView>(
+           {"cat", "tiger", "lion", "cat", "tiger", "lion", std::nullopt})});
+  assertQuery(op, expected);
+
+  // Test with array containing dictionary encoded elements.
+  size = 100;
+  BufferPtr offsets = AlignedBuffer::allocate<vector_size_t>(size, pool());
+  auto rawOffsets = offsets->asMutable<vector_size_t>();
+  std::iota(rawOffsets, rawOffsets + size, 0);
+  auto rowVector1 = makeRowVector(
+      {makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+       makeFlatVector<int32_t>(size, [](auto row) { return row; })});
+  auto dict1 = wrapInDictionary(offsets, rowVector1);
+
+  size = 200;
+  offsets = AlignedBuffer::allocate<vector_size_t>(size, pool());
+  rawOffsets = offsets->asMutable<vector_size_t>();
+  std::iota(rawOffsets, rawOffsets + size, 0);
+  auto rowVector2 = makeRowVector(
+      {makeFlatVector<double>(
+           size, [](auto row) { return row; }, nullEvery(7)),
+       makeFlatVector<StringView>(
+           size,
+           [](auto /* row */) { return StringView("abcd"); },
+           nullEvery(11))});
+  auto dict2 = wrapInDictionary(offsets, rowVector2);
+
+  vector =
+      makeRowVector({makeArrayVector({0}, dict1), makeArrayVector({0}, dict2)});
+  op = PlanBuilder()
+           .values({vector})
+           .unnest({}, {"c0", "c1"}, "ordinal")
+           .planNode();
+  expected = makeRowVector(
+      {makeFlatVector<int64_t>(
+           size, [](auto row) { return row; }, nullEvery(1, 100)),
+       makeFlatVector<int32_t>(
+           size, [](auto row) { return row; }, nullEvery(1, 100)),
+       makeFlatVector<double>(
+           size, [](auto row) { return row; }, nullEvery(7)),
+       makeFlatVector<StringView>(
+           size,
+           [](auto /* row */) { return StringView("abcd"); },
+           nullEvery(11)),
+       makeFlatVector<int64_t>(size, [](auto row) { return row + 1; })});
+  assertQuery(op, expected);
+
+  // Test with arrays of map and constant encoding.
+  using S = StringView;
+  using P = std::pair<int64_t, std::optional<S>>;
+
+  std::vector<P> a{P{1, S{"red"}}, P{2, S{"blue"}}, P{3, S{"green"}}};
```

Comment:
Maybe we should also try tests with Maps whose key, values are dictionary encoded vectors ?